asf_public_discourse_home_decarbonisation/utils/plotting_utils.py

```diff
@@ -0,0 +1,466 @@
+    plt.barh(ngram_labels, ngram_freqs, color=color)
+    plt.xlabel("Frequency")
+    plt.ylabel(f"{ngram_type.title()}s")
+    plt.title(f"Top {n} Most Common {ngram_type.title()}s")
+    plt.tight_layout()
+    plt.savefig(
+        os.path.join(
+            output_path, f"Top_{n}_Most_Common_{ngram_type.title()}s_horizontal.png"
+        )
+    )
+    plt.show()
+
+
+############ 6. Distribution of post lengths #############
+
+
+def plot_post_length_distribution(
+    dataframe, output_path=None, bins=50, color=NESTA_COLOURS[2]
+):
+    """
+    Plots and saves histograms of the post lengths in a given DataFrame.
+
+    Args:
+        dataframe (pandas.DataFrame): DataFrame containing the text data.
+        output_path (str, optional): Path to save the histogram images. If None, images are not saved.
+        bins (int, optional): Number of bins for the histogram. Default is 50.
+        color (str, optional): Color of the histogram bars. Default is 'blue'.
+
+    This function creates two histograms:
+    1. Standard scale histogram of post lengths.
+    2. Log scale histogram of post lengths.
+    """
+    # Calculate post length
+    dataframe["post_length"] = dataframe["text"].apply(lambda x: len(str(x).split()))
+
+    # Plot standard scale histogram
+    plt.figure(figsize=(12, 8))
+    sns.histplot(dataframe["post_length"], bins=bins, kde=False, color=color)
+    plt.title("Distribution of Post Lengths")
+    plt.xlabel("Post Length (Number of Words)")
+    plt.ylabel("Number of Posts")
+    plt.tight_layout()
+    if output_path:
+        plt.savefig(os.path.join(output_path, "distribution_of_post_lengths.png"))
+    plt.show()
+
+    # Plot log scale histogram
+    plt.figure(figsize=(12, 8))
+    sns.histplot(dataframe["post_length"], bins=bins, kde=False, color=color)
+    plt.title("Distribution of Post Lengths")
+    plt.xlabel("Post Length (Number of Words)")
+    plt.ylabel("Number of Posts (log scale)")
+    plt.yscale("log")
+    plt.tight_layout()
+    if output_path:
+        plt.savefig(
+            os.path.join(output_path, "distribution_of_post_lengths_logscale.png")
+        )
+    plt.show()
+
+
+############ 7. Frequency of selected keywords in posts using the dictionary ###########
+# Function for vertical bar chart plotting
+def plot_tag_vertical_bar_chart(dataframe, output_path, filename, df_threshold):
+    """
+    Plots and saves a vertical bar chart showing the frequency of selected tags in posts.
+
+    Args:
+        dataframe (pandas.DataFrame): DataFrame containing tag frequencies with columns "Tag" and "Frequency".
+        output_path (str): Directory path to save the output plot.
+        filename (str): Name of the file to save the plot.
+        df_threshold (int): Threshold frequency value for filtering tags.
+
+    This function creates a vertical bar chart where each bar represents a tag and its frequency.
+    Only tags with frequency greater than df_threshold are displayed.
```

Comment:
I would just shuffle the info around a little bit.
```suggestion
    Plots and saves a vertical bar chart showing the frequency of selected tags in posts.
    This function creates a vertical bar chart where each bar represents a tag and its frequency.
    Only tags with frequency greater than df_threshold are displayed.

    Args:
        dataframe (pandas.DataFrame): DataFrame containing tag frequencies with columns "Tag" and "Frequency".
        output_path (str): Directory path to save the output plot.
        filename (str): Name of the file to save the plot.
        df_threshold (int): Threshold frequency value for filtering tags.
```